cmd/gardenlet/app/gardenlet_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package app_test
+
+import (
+	"context"
+
+	. "github.com/gardener/gardener/cmd/gardenlet/app"
+	"github.com/gardener/gardener/pkg/apis/core"
+	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	"github.com/gardener/gardener/pkg/gardenlet/apis/config"
+	mockclient "github.com/gardener/gardener/pkg/mock/controller-runtime/client"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	gomegatypes "github.com/onsi/gomega/types"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Gardenlet", func() {
+	Describe("#CheckSeedConfig", func() {
+		var (
+			ctx    = context.TODO()
+			ctrl   *gomock.Controller
+			client *mockclient.MockClient
```

Comment:
Please use a fake client instead, see https://github.com/gardener/gardener/blob/master/docs/development/testing.md#writing-unit-tests:

> generally, prefer fakes over mocks, e.g., use controller-runtime fake client over mock clients

Reply:
Switched to fake client, but I do not see the big advantage in this scenario. It might be different in other cases.